src/ucp/core/ucp_ep.c

```diff
@@ -243,11 +244,12 @@ ucp_ep_adjust_params(ucp_ep_h ep, const ucp_ep_params_t *params)
         }
     }
 
-    if (params->field_mask & UCP_EP_PARAM_FIELD_ERR_HANDLER_CB) {
-        ucp_ep_setup_err_handler(ep, params->err_handler_cb);
+    if (params->field_mask & UCP_EP_PARAM_FIELD_ERR_HANDLER) {
+        ucp_ep_setup_err_handler(ep, &params->err_handler);
     }
 
-    ep->user_data = UCP_PARAM_VALUE(EP, params, user_data, USER_DATA, NULL);
+    ep->user_data = UCP_PARAM_VALUE(EP, params, user_data, USER_DATA,
```

Comment:
do we really want to override err_handler::arg with ep_params::user_data? if yes should document this in the API

Reply:
Otherwise we have to save both of them. It's documented: https://github.com/openucx/ucx/pull/2281/files#diff-b7d85f939678535440a859e57180b1c7R315